### DISCUSSION TITLE
A box with print-in-place hinges

### DISCUSSION THREAD

**Discussion by sgenoud** [sgenoud/replicad#53] [Show and tell]
I have been playing with print-in-place hinges and wanted to show how importing external code would look like, so here it is:

[My box with hinges](https://studio.replicad.xyz/share/https%3A%2F%2Fgist.githubusercontent.com%2Fsgenoud%2F8cd91468b28a05a3c1d8c0a964cb29a4%2Fraw%2Feea1c23755f3eca8fd1afbb97c6fd8e62a0e0e09%2Fbox-with-hinge.js) (unfortunately this does not work with Firefox 😢 - I am waiting for them to implement [worker modules](https://bugzilla.mozilla.org/show_bug.cgi?id=1247687))

<img width="735" alt="Capture d’écran 2022-07-21 à 10 02 33" src="https://user-images.githubusercontent.com/263325/180162282-21c00841-f35a-4d05-95ff-5ee795bf4540.png">

I would be curious if anyone would have some feedback on the design of the hinges. I am also wondering what a simple print in place closing mechanism should be for this kind of box

**Comment by paulftw** [comment]
This looks really cool. I'm a bit unsure about the overhangs seen in this picture - getting overhangs printed correctly can be fairly tricky, even with supports.

However, I've only tried printing hinges couple times: [this one](https://www.thingiverse.com/thing:4943713) worked really well on first attempt, and I only made it because I wanted to test a printer after repairs.

[This one](https://github.com/EtteGit/EnragedRabbitProject/blob/main/Carrot_Feeder/Stls/Gear%20box/%5Ba%5D_Side_Latch_x2.stl) failed during assembly after being printed with the same printer/filament/settings.
Maybe result was poor because I printed it with axis parallel to Z. That is weaker by design due to imperfect bond between filament layers.

I think most important feature of 3d printed hinges is how close to "invisible" they are when the lid is closed (or open, depending on the object). Replicad can really help with that since it can cut and modify a mesh created in some other software.

**Reply by sgenoud** [reply]
> This looks really cool. I'm a bit unsure about the overhangs seen in this picture - getting overhangs printed correctly can be fairly tricky, even with supports.

As far as I know 45 degrees overhangs are not an issue for printing - I have not printed that one, but one similar (the inside of the hinge also uses a 45 degrees overhang). That said there might be some equilibrium issues. I want to print a box using this technique this weekend - I will post photos here!

> However, I've only tried printing hinges couple times: [this one](https://www.thingiverse.com/thing:4943713) worked really well on first attempt, and I only made it because I wanted to test a printer after repairs.

My hinge design has the advantage of being able to print right sight up - which does not have the bridging of the calibration cube (which is not an issue for its side, but might for for bigger boxes. That said, I should see to improve my hinges library to make it easier to build it inside walls.

Have you encountered simple closing mechanisms for these kinds of simple, 3d printed boxes?

**Reply by raydeleu** [reply]
Your hinge design is interesting, but I think that having the rotation axis for the hinge below the lid only works for simple flat lids. As soon as you want to add a ridge so that the lid is not only held in place by the hinges, you will run into problems with this design. A solution could be to split the box into two halves. Here is an example: 

![image](https://user-images.githubusercontent.com/38007983/181383748-32be1d44-060e-46e6-a869-fe79c978f407.png)
(taken from https://www.printables.com/model/24004-hinged-box-with-latch) 

Notice that the design of the hinge is similar to yours. It also has a simple closing mechanism.
